src/application.rs

```diff
@@ -325,6 +327,14 @@ pub trait ApplicationHandler {
     fn memory_warning(&mut self, event_loop: &dyn ActiveEventLoop) {
         let _ = event_loop;
     }
+
+    /// Get the [`ApplicationHandler`] as mutable [`Any`].
+    ///
+    /// This is required to be implemented when platform specific traits are desired.
+    #[inline(always)]
+    fn as_any(&mut self) -> Option<&mut dyn Any> {
+        None
+    }
```

Comment:
I played around a bit to figure out how we can get rid of this:
https://github.com/daxpedda/winit/commit/374014880b0606af9796a0ecfaff00f31a3312e5

Mostly inspired by [`upcast`](https://crates.io/crates/upcast).
I'm warming up to the VTable idea, but I couldn't find any documentation on how stable this is, so I'm unsure what the proposal around this was exactly.

Reply:
>I'm warming up to the VTable idea, but I couldn't find any documentation on how stable this is, so I'm unsure what the proposal around this was exactly.

vtable is only for completely out of tree stuff, for regular staff you don't need vtables, since users will have `Option<dyn Handler>` and you can just `if let` it. But function pointers are completely safe and stable, there's nothing wrong with them and they are all completely safe code.

it's just in the current state of things I decided with vtable, since it's a bit more reliable, but we can already do other approach actually.

So, we'd have `-> Option<&mut dyn MacOSApplicationHandler>` from the `ApplicationHandler` itself.

Reply:
Yeah okay, I was thinking of totally different things.
I would like to avoid having to use `register_x()` entirely. That is possible by *unsafely* reading the VTable and figuring out at runtime if a trait is implemented for a certain type.